src/adb.js

```diff
@@ -1 +1,353 @@
+	 */
+	stopApp(deviceId, appid) {
+		return this.getPid(deviceId, appid)
+			.then(pid => {
+				console.log(pid);
+				return this.shell(deviceId, 'am force-stop ' + appid);
+			});
+	}
+
+	/**
+	 * Retrieves a list of connected devices and emulators.
+	 * @returns {Promise}
+	 */
+	devices() {
+		return new Connection(this)
+			.exec('host:devices')
+			.then(data => {
+				return parseDevices(this, data);
+			});
+	}
+
+	/**
+	 * Streams output from logcat into the specified handler until the adb logcat
+	 * process ends.
+	 * @param {String} deviceId - The id of the device or emulator
+	 * @param {Function} handler - A function to call whenever data becomes available
+	 * @returns {Promise}
+	 */
+	logcat(deviceId, handler) {
+		return sdk
+			.detect(this.config)
+			.then(result => {
+				return new Promise((resolve, reject) => {
+					const args = [
+						'-s',
+						deviceId,
+						'logcat'
+						//'-v', 'brief', '-b', 'main'
+					];
+					const child = spawn(result.sdk.executables.adb, args);
+
+					child.stdout.on('data', data => {
+						handler(data.toString());
+					});
+
+					child.on('close', () => resolve());
+				});
+			});
+	}
+}
+
+/**
+ * Parses the device list, then fetches additional device info.
+ * @param {ADB} adb - The ADB instance
+ * @param {Buffer|String} data - The buffer containing the list of devices
+ */
+function parseDevices(adb, data) {
+	let devices = [];
+	const emuMgr = new EmulatorManager(adb.config);
+
+	(data || '').toString().split('\n').forEach(item => {
+		let p = item.split(/\s+/);
+		if (p.length > 1) {
+			devices.push({
+				id: p.shift(),
+				state: p.shift()
+			});
+		}
+	});
+
+	return Promise
+		.all(devices.map((device, index) => {
+			const deviceId = device.id;
+			return adb
+				.shell(deviceId, 'cat /system/build.prop')
+				.then(data => {
+					let info = {};
+					data.toString().split('\n').forEach(line => {
+						const p = line.indexOf('=');
+						if (p !== -1) {
+							const key = line.substring(0, p).trim();
+							const value = line.substring(p + 1).trim();
+							switch (key) {
+								case 'ro.product.model.internal':
+									info.modelnumber = value;
+									break;
+								case 'ro.build.version.release':
+								case 'ro.build.version.sdk':
+								case 'ro.product.brand':
+								case 'ro.product.device':
+								case 'ro.product.manufacturer':
+								case 'ro.product.model':
+								case 'ro.product.name':
+									info[key.split('.').pop()] = value;
+									break;
+								case 'ro.genymotion.version':
+									info.genymotion = value;
+									break;
+								default:
+									if (key.indexOf('ro.product.cpu.abi') == 0) {
+										Array.isArray(info.abi) || (info.abi = []);
+										value.split(',').forEach(abi => {
+											abi = abi.trim();
+											if (info.abi.indexOf(abi) === -1) {
+												info.abi.push(abi);
+											}
+										});
+									}
+									break;
+							}
+						}
+					});
+					return info;
```

Comment:
Instead of adding "build.props" to an `info` object and mixing into the `device` object, just set these props directly on the `device` object.

No need to return anything from this function.